### PR TITLE
[Network]: get virtualnetwork gateway vpn connection health API

### PR DESCRIPTION
**Pull request by ninweninwe**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
Should this be targeted to `network-april-release` (with API version 2019-04-01)? @MikhailTryakhov

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json [hunk 1]
```diff
@@ -1572,7 +1572,55 @@
           "nextLinkName": "nextLink"
         }
       }
-    }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/getvpnclientconnectionhealth": {
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json]:
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/getVpnclientConnectionHealth": {
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json [hunk 2]
```diff
@@ -2775,6 +2823,90 @@
...
+        "vpnUserName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The route's AS path sequence"
+        },
+        "maxBandwidth": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "The route's weight"
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json]: please check this description and other for the properties below

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json [hunk 3]
```diff
@@ -2775,6 +2823,90 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VpnClientConnectionHealthDetail"
+          },
+          "description": "List of vpn client connection health"
+        }
+      },
+      "description": "List of virtual network gateway vpn client connection health"
+    },
+    "VpnClientConnectionHealthDetail": {
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json]: Shouldn't this be plural, "Detail **s** "?

**Comment by ninweninwe** [comment]
Any one knows why azure sdk for net build failed?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json (hunk 3 above)
Comment (ninweninwe) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json]: no "s".  the above array is an array of VpnClientConnectionHealthDetail

**Comment by MikhailTryakhov** [comment]
@number213 it should. and newer api version is ```2019-04-01```

**Comment by ninweninwe** [comment]
> Should this be targeted to `network-april-release` (with API version 2019-04-01)? @MikhailTryakhov

Moved to 2019-04-01

**Comment by dsgouda** [comment]
@MikhailTryakhov is this the right target branch? Shouldn't it be the april release branch?

**Comment by ninweninwe** [comment]
> @MikhailTryakhov is this the right target branch? Shouldn't it be the april release branch?

@dsgouda I moved it to april release folder

**Comment by ninweninwe** [comment]
need to work on april release branch cancel this PR

**Pull request closed without merging**